python/ray/serve/_private/autoscaling_state.py

```diff
@@ -443,28 +427,15 @@ def record_request_metrics_for_replica(
                 send_timestamp=send_timestamp,
             )
 
-    def record_request_metrics_for_handle(
-        self,
-        *,
-        deployment_id: str,
-        handle_id: str,
-        actor_id: Optional[str],
-        handle_source: DeploymentHandleSource,
-        queued_requests: float,
-        running_requests: Dict[ReplicaID, float],
-        send_timestamp: float,
-    ) -> None:
+    def record_request_metrics_for_handle(self, report: HandleMetricReport) -> None:
         """Update request metric for a specific handle."""
 
-        if deployment_id in self._autoscaling_states:
-            self._autoscaling_states[deployment_id].record_request_metrics_for_handle(
-                handle_id=handle_id,
-                actor_id=actor_id,
-                handle_source=handle_source,
-                queued_requests=queued_requests,
-                running_requests=running_requests,
-                send_timestamp=send_timestamp,
-            )
+        try:
+            autoscaling_state = self._autoscaling_states[report.deployment_id]
+        except KeyError:
+            return
```

Comment:
Totally personal preference, but I find relying on exceptions to handle in a set check makes the code a bit obscure to the reader. I prefer the existing implementation.